src/Controller/ModulesController.php

```diff
@@ -543,6 +557,33 @@ public function relationshipsJson($id, string $relation): void
         $this->set('_serialize', array_keys($response));
     }
 
+    /**
+     * Retrieve URL to get objects available for a relation
+     *
+     * @param string $relation The relation name.
+     * @return string
+     */
+    protected function availableRelationshipsUrl(string $relation): string
+    {
+        $defaults = [
+            'children' => '/objects',
+            'parent' => '/folders',
+            'parents' => '/folders',
+        ];
+        $defaultUrl = (string)Hash::get($defaults, $relation);
+        if (!empty($defaultUrl)) {
+            return $defaultUrl;
+        }
+
+        $relationsSchema = $this->Schema->getRelationsSchema();
+        $types = $this->Modules->relatedTypes($relationsSchema, $relation);
+        if (count($types) === 1) {
+            return sprintf('/%s', $types[0]);
+        }
+
+        return '/objects?filter[type][]=' . implode('&filter[type][]=', $types);
```

Comment:
if there are several types, url can be very long. I wonder wether BE api allow a shorter version of this kind of url, like:
`/objects?filter[type]=<comma separated types>`

Reply:
the alternative format you suggested is not supported right now by the API, but could be an improvement surely...